app/BUILD.bazel

```diff
@@ -258,16 +349,55 @@ in order to keep Gradle building.
  )
 for file in ALTERED_VIEWS]
 
+# Binding Adapters
 
 # TODO(#1617): Remove genrules post-gradle
-# Adapter files altered by genrules to be excluded in the app library.
+'''
+Binding adapter files under the org/oppia/app/databinding folder that need to be altered by the
+databinding binding adapter genrule.
+
+Please note that binding adapters have been split up into two types based on their location:
+databinding_adapters can be found under org/oppia/app/databinding and recyclerview_adapters can be
+found under org/oppia/app/recyclerview. Note that a genrule has not been created for
+recyclerview_adapters as it is not currently needed.
+
+Place your file here if:
+ - It is a binding adapter
+ - It imports resources (ex: `import org.oppia.app.R`)
+ - It is filed under org/oppia/app/databinding
+
+DO NOT place your file here if:
+ - It depends on layout files
+ - It does not import any resources
+ - It is found under org/oppia/app/recyclerview
+ - It uses Kotlin synthetic imports
+'''
 ALTERED_DATABINDING_ADAPTERS = [
     "src/main/java/org/oppia/app/databinding/DrawableBindingAdapters.java",
     "src/main/java/org/oppia/app/databinding/ImageViewBindingAdapters.java",
     "src/main/java/org/oppia/app/databinding/TextViewBindingAdapters.java",
 ]
 
-# Source files and genrule targets for databinding_adapters library.
+'''
+Binding adapter files under the org/oppia/app/databinding folder that DO NOT need to be altered by
+the databinding binding adapter genrule.
+
+Please note that binding adapters have been split up into two types based on their location:
+databinding_adapters can be found under org/oppia/app/databinding and recyclerview_adapters can be
+found under org/oppia/app/recyclerview. Note that a genrule has not been created for
+recyclerview_adapters as it is not currently needed.
+
+Place your file here if:
+ - It is a binding adapter
+ - It does not import resources
+ - It is filed under org/oppia/app/databinding
+
+DO NOT place your file here if:
+ - It depends on layout files
+ - It imports resources
+ - It is found under org/oppia/app/recyclerview
+ - It uses Kotlin synthetic imports
+'''
 DATABINDING_ADAPTERS = [
     "src/main/java/org/oppia/app/databinding/EditTextBindingAdapters.java",
```

Comment:
Suggest also mentioning for adapter documentations that adapters _have_ to be defined in one of the adapter lists, and they have to be in Java.

Reply:
Ok